File: 58A.py
s = input()
orig = 'hello'
i = 0
j = 0
l = len(s)
while i<l:
    if s[i] == orig[j]:
        j = j+1
    i = i+1
    if(j==5):
        break
if j==5:
    print('YES')
else:
    print('NO')

File: 58A.java
import java.util.Scanner;

class FiftyEightA {

    public static void solve(String str) {
        char hello[] = {'h','e','l','l','o'};
        char strArr[] = str.toCharArray();

        int i,j;
        i = j = 0;

        while(j < strArr.length) {
            if(strArr[j] == hello[i]) {
                i++;
            }
            if( i == hello.length) {
                break;
            }
            j++;
        }
        if(i == hello.length) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }

    public static void main(String []args) {
        Scanner sc = new Scanner(System.in);

        String str = sc.next();

        solve(str);

        sc.close();
    }
    
}
